packages/shared/src/hooks/useReadingRank.ts

```diff
@@ -42,26 +42,22 @@ const checkShouldShowRankModal = (
   neverShowRankModal?: boolean,
 ) => {
   if (neverShowRankModal !== null && neverShowRankModal !== undefined) {
-    return neverShowRankModal;
+    return !neverShowRankModal;
   }
 
   if (rankLastSeen === undefined) {
     return true;
   }
 
-  if (rankLastSeen === null) {
-    return false;
-  }
-
-  if (!lastReadTime) {
+  if (!rankLastSeen) {
```

Comment:
Why if it's `null` or `false` we need to not show the rank modal?
In what cases the `rankLastSeen` can be `null` or `false`?

Reply:
The value `null` is our default value `onload` while we are still fetching the data of `rankLastSeen`, we don't want to accidentally show it - the reason for `false` is the way we pass the value for `rankLastSeen` like the snippet below. So when the value is not yet loaded (`loadedAlerts`) it will return false but if it is fetched already, it should give the actual value from `alerts.rankLastSeen` (be it `undefined` or the actual `Date`).

With that, I have noticed a different unexpected behavior related to this. Let me check that one out. 

https://github.com/dailydotdev/apps/blob/3e17c396312d2fe7b478866358cc92a716b7e730/packages/shared/src/hooks/useReadingRank.ts#L95

Reply:
If `rankLastSeen` is null while loading then why do we need `loadedAlerts` for?

Reply:
That is so we can indicate that the `null` value was not the default one but sent from the remote settings since our DB value is nullable. If it is `null` because of the default values, we should not show anything yet.

For this to be much clearer, let me take the `loadedAlerts` inside the function for easier readability and maintainability.